lib/remi/data_subjects/gsheet_file.rb

```diff
@@ -0,0 +1,163 @@
+  require 'googleauth'
+  require 'googleauth/stores/file_token_store'
+  require 'daru'
+  require 'fileutils'
+
+  # Contains methods shared between Salesforce Extractor/Parser/Encoder/Loader
+  class Extractor::Gsheet < Extractor::FileSystem
+
+    def initialize(*args, **kargs, &block)
+      super
+      init_gsheet_extractor(*args, **kargs)
+    end
+
+    attr_reader :folder_name
+    attr_reader :itk_file
+    attr_reader :data
+
+
+    def authorize()
+
+      FileUtils.mkdir_p(File.dirname(@credentials_path))
+      client_id   = Google::Auth::ClientId.from_file(@client_secrets_path)
+      token_store = Google::Auth::Stores::FileTokenStore.new(file: @credentials_path)
+      authorizer  = Google::Auth::UserAuthorizer.new(client_id, [Google::Apis::DriveV3::AUTH_DRIVE, Google::Apis::SheetsV4::AUTH_SPREADSHEETS], token_store)
+      user_id     = 'default'
+      creds       = authorizer.get_credentials(user_id)
+      if creds.nil?
+        raise "Error, Credentials file does not exist"
+      end
+      creds
+    end
+
+
+    def get_file_list(folder_id)
+
+      service                                 = Google::Apis::DriveV3::DriveService.new
+      service.client_options.application_name = @application_name
+      service.authorization                   = authorize()
+      response = service_list_files(service, folder_id)
+      response.files
+
+    end
+
+    def service_list_files(service, folder_id)
+      service.list_files(q: "'#{folder_id}' in parents", page_size: 10, order_by: 'createdTime desc', fields: 'nextPageToken, files(id, name, createdTime, mimeType)')
+    end
+
+    def get_spreadsheet_vals(service, spreadsheet_id)
+      service.get_spreadsheet_values(spreadsheet_id, 'Sheet1')
+    end
+
+    def extract()
+
+      service                                 = Google::Apis::SheetsV4::SheetsService.new
+      service.client_options.application_name = @application_name
+      service.authorization                   = authorize()
+      @data                                   = []
+      file_list  = get_file_list(@default_folder_id)
+
+      file_list.each do |file|
+        file = file.to_h
+        response = get_spreadsheet_vals(service, file[:id])
+        data.push(response)
+      end
+
+      self
+
+    end
+
+    # @return [Array<Extractor::FileSystemEntry>] (Memoized) list of objects in the bucket/prefix
+    def all_entries
+      @all_entries ||= all_entries!
+    end
+
+    # @return [Array<Extractor::FileSystemEntry>] (Memoized) list of objects in the bucket/prefix
+    def all_entries!
+
+      gsheet_entries = get_file_list(@default_folder_id)
+      gsheet_entries.map do |entry|
+        entry = entry.to_h
+        FileSystemEntry.new(
+          pathname:       File.join(@default_folder_id, entry[:name]),
+          create_time:    entry[:created_time],
+          modified_time:  entry[:created_time]
+        )
+      end
+    end
+
+    private def init_gsheet_extractor(*args, credentials:, **kargs)
```

Comment:
Ruby convention is to declare a private section (https://github.com/bbatsov/ruby-style-guide#indent-public-private-protected) instead of prefacing each private def with `private`.  So,

````
...
private

def init_gsheet_extractor(...
````